Assets/Solana Integration/On-Chain/SolChessClient.cs

```diff
@@ -354,4 +413,11 @@ private static int UnMapRank(int rank)
 
     #endregion
 
+    # region Session
+    private string deriveSessionPassword()
+    {
+        return $"{Web3.Account.PublicKey}!Cv%{sessionPassword}";
```

Comment:
Perhaps using an hash function such as Sha256 will make it slightly more secure, also combining more parameters could reduce the validity/scope of the password. E.g. something like: sha256(publicKey + sessionPassword + Application.Platform + startTimestamp? )